File: src/main/java/com/example/backend/service/TarotService.java
package com.example.backend.service;

import com.example.backend.dto.tarot.TarotCardDto;
import com.example.backend.dto.tarot.TarotRequest;
import com.example.backend.dto.tarot.TarotResponse;
import java.util.List;
import java.util.UUID;

public interface TarotService {

    List<TarotCardDto> getAllCards();

    TarotResponse ask(TarotRequest request) throws Exception;
    UUID askAsync(TarotRequest request) throws Exception;

    TarotCardDto randomOne();
}

File: src/main/kotlin/com/example/backend/service/TarotService.kt
package com.example.backend.service

import com.example.backend.dto.tarot.TarotCardDto
import com.example.backend.dto.tarot.TarotRequest
import com.example.backend.dto.tarot.TarotResponse
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface TarotService {

    fun getAllCards(): Flux<TarotCardDto?>

    fun ask(request: TarotRequest?): Mono<TarotResponse?>
}